Source/WebKit/UIProcess/Cocoa/WKScreenTimeConfigurationObserver.mm

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "WKScreenTimeConfigurationObserver.h"
+
+#if ENABLE(SCREEN_TIME)
+
+#import "WKWebViewInternal.h"
+#import <wtf/RetainPtr.h>
+#import <wtf/WeakObjCPtr.h>
+#import <wtf/darwin/DispatchExtras.h>
+
+#import <pal/cocoa/ScreenTimeSoftLink.h>
+
+static void *screenTimeConfigurationObserverKVOContext = &screenTimeConfigurationObserverKVOContext;
+
+@implementation WKScreenTimeConfigurationObserver {
+    RetainPtr<STScreenTimeConfigurationObserver> _screenTimeConfigurationObserver;
+    WeakObjCPtr<WKWebView> _webView;
+}
+
+- (instancetype)initWithView:(WKWebView *)view
+{
+    if (!(self = [super init]))
+        return nil;
+
+    _webView = view;
+
+    return self;
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary<NSKeyValueChangeKey, id> *)change context:(void *)context
+{
+    if (context == &screenTimeConfigurationObserverKVOContext) {
+        ensureOnMainRunLoop([webView = _webView] {
+            [webView _updateScreenTimeBasedOnWindowVisibility];
+        });
+        return;
+    }
+
+    [super observeValueForKeyPath:keyPath ofObject:object change:change context:context];
+}
+
+- (void)startObserving
+{
+    _screenTimeConfigurationObserver = adoptNS([PAL::allocSTScreenTimeConfigurationObserverInstance() initWithUpdateQueue:globalDispatchQueueSingleton(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)]);
```

Comment:
I know we only create `WKScreenTimeConfigurationObserver` if one doesn't exist, but maybe same idea with `STScreenTimeConfigurationObserver`? In case the caller does something redundant.

```objc
    if (!_screenTimeConfigurationObserver) {
        _screenTimeConfigurationObserver = ...
    }
    ...
```